### PR TITLE
Use AvlSetInt instead of lists for inc. matrix

### PR DESCRIPTION
**Pull request by sjoelund**
The incidence matrix takes a long time to calculate for certain systems,
especially when you have one equation that ties into all of the other
equations.

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-07-19_16-11-29](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3048/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-07-28_09-06-00](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3066/).

**Comment by sjoelund** [comment]
@vruge sadly, Pendulum still does not work and the heat exchanger gets a timeout.

File: Compiler/BackEnd/BackendDAEUtil.mo [hunk 1]
```diff
@@ -2928,33 +2945,31 @@ algorithm
     case (BackendDAE.VAR(varKind = BackendDAE.STATE(derName=SOME(_)))::rest,i::irest,_,_)
       equation
         i1 = if intGe(diffindex,1) then i else -i;
-        b = List.isMemberOnTrue(i1, inVarIndxLst, intEq);
-        vars = List.consOnTrue(not b, i1, inVarIndxLst);
+        vars = AvlSetInt.add(inVarIndxLst, i1);
```
Comment (hkiel) [inline Compiler/BackEnd/BackendDAEUtil.mo]: @sjoelund the old code only added conditionally. Is this change intended?

File: Compiler/BackEnd/BackendDAEUtil.mo [hunk 2]
```diff
@@ -2928,33 +2945,31 @@ algorithm
...
       then incidenceRowExp1(rest,irest,vars,diffindex);
     case (BackendDAE.VAR(varKind = BackendDAE.STATE(index=diffidx))::rest,i::irest,_,_)
       equation
         i1 = if intGe(diffindex,diffidx) then i else -i;
-        b = List.isMemberOnTrue(i1, inVarIndxLst, intEq);
-        vars = List.consOnTrue(not b, i1, inVarIndxLst);
+        vars = AvlSetInt.add(inVarIndxLst, i1);
```
Comment (hkiel) [inline Compiler/BackEnd/BackendDAEUtil.mo]: same here

File: Compiler/BackEnd/BackendDAEUtil.mo [hunk 3]
```diff
@@ -2928,33 +2945,31 @@ algorithm
...
       then incidenceRowExp1(rest,irest,vars,diffindex);
     case (_::rest,i::irest,_,_)
       equation
-        b = List.isMemberOnTrue(i, inVarIndxLst, intEq);
-        vars = List.consOnTrue(not b, i, inVarIndxLst);
+        vars = AvlSetInt.add(inVarIndxLst, i);
```
Comment (hkiel) [inline Compiler/BackEnd/BackendDAEUtil.mo]: and here!

**Comment by lochel** [comment]
@sjoelund I'll take a look.

**Comment by lochel** [comment]
@wibraun I have solved the conflicts in f414416f32f1e52451ba2b1184824ec2647adad8.
And this is the recent report: [OpenModelica_TEST_PULL_REQUEST 2016-10-26](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3311/)

**Comment by casella** [comment]
@sjoelund are you going to open another PR with your changes?

**Comment by sjoelund** [comment]
No, I was just cleaning my stale branches in github and this got deleted by accident...

**Comment by lochel** [comment]
The problem is that we do not consider dynamic state selection within the initialization process.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-11-18_09-16-13](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3427/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-11-18_12-06-28](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3428/).

**Comment by sjoelund** [comment]
@casella @rfranke: I guess we should postpone merging this PR into master until after the 1.11 branch? It does change things from before (the Pendulum model stops working for example).

**Comment by casella** [comment]
Absolutely!

**Comment by hkiel** [comment]
When does 1.11 come?

**Comment by sjoelund** [comment]
Before 1.10 :)

**Comment by casella** [comment]
The plan is to branch in on Friday 25th

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-11-29_14-26-30](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3476/).

**Comment by casella** [comment]
@sjoelund can you proceed with the PR?

**Comment by casella** [comment]
I guess you have to change the reference results to get this through

**Comment by sjoelund** [comment]
Yes. I've been busy teaching and preparing the release instead :)

**Comment by sjoelund** [comment]
@wibraun : Please rewrite testDASSLnoEquidistant; I will move it to failing tests

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2016-11-30_13-04-06](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3487/console).
